tests/drivers/gpio/gpio_more_loops/boards/nrf54h20dk_nrf54h20_cpuapp.overlay

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/ {
+	/* Test requirements:
+	 * out-gpios[0] wire connected with in-gpios[0],
+	 * out-gpios[1] wire connected with in-gpios[1],
+	 * etc.
+	 * Output-input GPIO pair must have identical active level flag.
+	 */
+	test_gpios {
+		compatible = "gpio-leds";
+		out_gpios: out_gpios {
+			gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>, <&gpio2 8 GPIO_ACTIVE_HIGH>,
+					<&gpio1 2 GPIO_ACTIVE_HIGH>, <&gpio7 2 GPIO_ACTIVE_HIGH>,
+					<&gpio7 6 GPIO_ACTIVE_HIGH>;
+		};
+
+		in_gpios: in_gpios {
+			gpios = <&gpio0 1 GPIO_ACTIVE_HIGH>, <&gpio2 9 GPIO_ACTIVE_HIGH>,
+					<&gpio1 3 GPIO_ACTIVE_HIGH>, <&gpio1 0 GPIO_ACTIVE_HIGH>,
+					<&gpio7 7 GPIO_ACTIVE_HIGH>;
+		};
+	};
```

Comment:
I would use `zephyr,user` rather than a re-purposed compatible like "gpio-leds" here for full flexibility and independence from a led binding :) Something like:
```
	zephyr,user {
		out-gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>, <&gpio2 8 GPIO_ACTIVE_HIGH>,
			    <&gpio1 2 GPIO_ACTIVE_HIGH>, <&gpio7 2 GPIO_ACTIVE_HIGH>,
			    <&gpio7 6 GPIO_ACTIVE_HIGH>;
		};

		in-gpios = <&gpio0 1 GPIO_ACTIVE_HIGH>, <&gpio2 9 GPIO_ACTIVE_HIGH>,
			    <&gpio1 3 GPIO_ACTIVE_HIGH>, <&gpio1 0 GPIO_ACTIVE_HIGH>,
			    <&gpio7 7 GPIO_ACTIVE_HIGH>;
	};
```
and
```
static const struct gpio_dt_spec pin = GPIO_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), out_gpios, 0);
```

Reply:
I applied this suggestion (see second commit in this PR).
But test started to fail on the first compare. Which I don't understand.
```
*** Booting nRF Connect SDK v2.7.99-f932aecfc006 ***
*** Using Zephyr OS v3.7.99-2cac88cbaff3 ***
Running TESTSUITE gpio_loopbacks
===================================================================
Test executed on nrf54h20dk@0.9.0/nrf54h20/cpuapp
Exisitng GPIO Loops:
GPIO Loop 1
IN: gpio@938000, 1;	OUT: gpio@938000, 0
GPIO Loop 2
===================================================================
START - test_gpios_are_working

    Assertion failed at WEST_TOPDIR/nrf/tests/drivers/gpio/gpio_more_loops/src/main.c:65: _check_inputs: (current not equal to expected)
IN_1 = 0, while expected 1
 FAIL - test_gpios_are_working in 0.022 seconds
```
Are You able to find error?

Reply:
After investigation I found that:
- definition of test GPIOs under zepyr,user is OK for nrf54l15dk;
- definition of test GPIOs under zepyr,user fails for nrf54h20dk -> nrf-regtool doesn't assign GPIO ownership in the UICR to the Application core.

Reply:
I drop this change because I don't know what mechanism is used by nrf-regtool to correctly assign GPIO ownership. Definitely, nrf-regtool works fine with LED binding.